Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewCell.cs

```diff
@@ -0,0 +1,1265 @@
+        // </returns>
+        // <param name="rowIndex">The zero-based index of the row containing the cell.</param><param name="firstCell">true to indicate that the cell is in the first column of the region defined by the selected cells; otherwise, false.</param><param name="lastCell">true to indicate that the cell is the last column of the region defined by the selected cells; otherwise, false.</param><param name="inFirstRow">true to indicate that the cell is in the first row of the region defined by the selected cells; otherwise, false.</param><param name="inLastRow">true to indicate that the cell is in the last row of the region defined by the selected cells; otherwise, false.</param><param name="format">The current format string of the cell.</param><exception cref="T:System.ArgumentOutOfRangeException"><paramref name="rowIndex"/> is less than 0 or greater than or equal to the number of rows in the control.</exception><exception cref="T:System.InvalidOperationException">The value of the cell's <see cref="P:System.Windows.Forms.DataGridViewElement.DataGridView"/> property is null.-or-<see cref="P:System.Windows.Forms.DataGridViewCell.ColumnIndex"/> is less than 0, indicating that the cell is a row header cell.</exception><exception cref="T:System.Exception">Formatting failed and either there is no handler for the <see cref="E:System.Windows.Forms.DataGridView.DataError"/> event of the <see cref="T:System.Windows.Forms.DataGridView"/> control or the handler set the <see cref="P:System.Windows.Forms.DataGridViewDataErrorEventArgs.ThrowException"/> property to true. The exception object can typically be cast to type <see cref="T:System.FormatException"/>.</exception>
+        // protected virtual object GetClipboardContent(int rowIndex, bool firstCell, bool lastCell, bool inFirstRow, bool inLastRow, string format)
+        
+        /// <summary>
+        /// Returns the bounding rectangle that encloses the cell's content area using a default <see cref="T:System.Drawing.Graphics"/> and cell style currently in effect for the cell.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The <see cref="T:System.Drawing.Rectangle"/> that bounds the cell's contents.
+        /// </returns>
+        /// <param name="rowIndex">The index of the cell's parent row.</param><exception cref="T:System.ArgumentOutOfRangeException">The specified <paramref name="rowIndex"/> is less than 0 or greater than the number of rows in the control minus 1. </exception><exception cref="T:System.InvalidOperationException"><see cref="P:System.Windows.Forms.DataGridViewCell.ColumnIndex"/> is less than 0, indicating that the cell is a row header cell.</exception>
+        public Rectangle GetContentBounds(int rowIndex)
+        {
+            Contract.Requires(this.DataGridView != null);
+            Contract.Requires(rowIndex >= 0 || rowIndex < this.DataGridView.Rows.Count);
```

Comment:
And I can't see this arg check in this method as well...

In default implementation I see this:

```
protected virtual Rectangle GetContentBounds(Graphics graphics, DataGridViewCellStyle cellStyle, int rowIndex)
        {
            return Rectangle.Empty;
        }
```

So I don't think we can force this contract here.